File: SoftInvModel/src/main/java/pe/edu/pucp/softinv/model/Personas/UsuarioDTO.java
package pe.edu.pucp.softinv.model.Personas;

import pe.edu.pucp.softinv.model.Servicio.CitaDTO;

import java.util.ArrayList;

//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
public abstract class UsuarioDTO {
    private String nombre;
    private String Primerapellido;
    private String Segundoapellido;
    private String correoElectronico;
    private String contrasenha;
    private String celular;
    private String urlFotoPerfil;
    protected String rol;
    private ArrayList<CitaDTO> citas;
    private Integer idUsuario;

    public abstract void setRol();

    public String getRol(){
        return rol;
    }

    public ArrayList<CitaDTO> getCitas() {
        return citas;
    }

    public Integer getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(Integer idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getUrlFotoPerfil() {
        return urlFotoPerfil;
    }

    public void setUrlFotoPerfil(String urlFotoPerfil) {
        this.urlFotoPerfil = urlFotoPerfil;
    }

    public UsuarioDTO() {
        nombre = null;
        Primerapellido = null;
        Segundoapellido = null;
        rol=null;
        contrasenha = null;
        correoElectronico = null;
        celular = null;
        urlFotoPerfil = null;
        citas=null;
        idUsuario=null;
    }

    public UsuarioDTO(String nombre,String PrimerApellido, String SegundoApellido, String correoElectronico, String contrasenha, String celular, String urlFotoPerfil,Integer idUsuario) {
        this.nombre = nombre;
        this.Primerapellido = PrimerApellido;
        this.Segundoapellido = SegundoApellido;
        this.correoElectronico = correoElectronico;
        this.contrasenha = contrasenha;
        this.celular = celular;
        this.urlFotoPerfil = urlFotoPerfil;
        this.idUsuario = idUsuario;
    }

    public String getPrimerapellido() {
        return Primerapellido;
    }

    public void setPrimerapellido(String primerapellido) {
        Primerapellido = primerapellido;
    }

    public String getSegundoapellido() {
        return Segundoapellido;
    }

    public void setSegundoapellido(String segundoapellido) {
        Segundoapellido = segundoapellido;
    }

    public String getCorreoElectronico() {
        return correoElectronico;
    }

    public void setCorreoElectronico(String correoElectronico) {
        this.correoElectronico = correoElectronico;
    }

    public String getContrasenha() {
        return contrasenha;
    }

    public void setContrasenha(String contrasenha) {
        this.contrasenha = contrasenha;
    }

    public String getCelular() {
        return celular;
    }

    public void setCelular(String celular) {
        this.celular = celular;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

}

File: C#/SoftInvModel/Personas/UsuarioDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace softinv.model
{
    public abstract class UsuarioDTO
    {
        private string nombre;
        private string Primerapellido;
        private string Segundoapellido;
        private string correoElectronico;
        private string contrasenha;
        private string celular;
        private string urlFotoPerfil;
        protected string rol;
        private BindingList<CitaDTO> citas;
        private int idUsuario;

        public abstract void setRol();

        public string getRol()
        {
            return rol;
        }

        public UsuarioDTO()
        {
            nombre = null;
            Primerapellido = null;
            Segundoapellido = null;
            rol = null;
            contrasenha = null;
            correoElectronico = null;
            celular = null;
            urlFotoPerfil = null;
            citas = null;
        }

        public UsuarioDTO(string nombre, string PrimerApellido, string SegundoApellido, string correoElectronico, string contrasenha, string celular, string urlFotoPerfil, int idUsuario)
        {
            this.nombre = nombre;
            this.Primerapellido = PrimerApellido;
            this.Segundoapellido = SegundoApellido;
            this.correoElectronico = correoElectronico;
            this.contrasenha = contrasenha;
            this.celular = celular;
            this.urlFotoPerfil = urlFotoPerfil;
            this.idUsuario = idUsuario;
        }

        public string Nombre { get => nombre; set => nombre = value; }
        public string Primerapellido1 { get => Primerapellido; set => Primerapellido = value; }
        public string Segundoapellido1 { get => Segundoapellido; set => Segundoapellido = value; }
        public string CorreoElectronico { get => correoElectronico; set => correoElectronico = value; }
        public string Contrasenha { get => contrasenha; set => contrasenha = value; }
        public string Celular { get => celular; set => celular = value; }
        public string UrlFotoPerfil { get => urlFotoPerfil; set => urlFotoPerfil = value; }
        internal BindingList<CitaDTO> Citas { get => citas; set => citas = value; }
        public int IdUsuario { get => idUsuario; set => idUsuario = value; }

    }
}